wsclient/src/main/java/org/hpccsystems/ws/client/platform/EclInfo.java

```diff
@@ -0,0 +1,70 @@
+package org.hpccsystems.ws.client.platform;
+
+import java.util.HashMap;
+
+import org.hpccsystems.ws.client.utils.FileFormat;
+
+public class EclInfo
+{
+    private String originalEcl;
+
+    public EclInfo()
+    {
+    }
+
+    public EclInfo(DFURecordDefInfo recs)
+    {
+        this.setRecordsets(new HashMap<String, DFURecordDefInfo>());
+        this.getRecordsets().put(EclInfo.UNNAMED, recs);
+    }
+
+    public HashMap<String, DFURecordDefInfo> recordsets = new HashMap<String, DFURecordDefInfo>();
```

Comment:
Setting it to an empty array simplifies checks. I can just check size ==0 instead of obj==null || size==0 every time I check the object for recordsets.